### ISSUE TITLE
tracksToReaperProject should generate vst2 state + vst2 parameter automation

### ISSUE THREAD

**Issue by CharlesHolbrow** [CharlesHolbrow/cybr#83]
# Objectives:
1. VST2 State handling in
  - `tracksToReaperProject` should correctly output vst2 state info by instantiating the plugin in the fluid engine server. 
  -  If a plugin has parameters specified in its `.parameters` field, `tracksToReaperProject` should instantiate an instance of the plugin, set any parameter (using their normalized values) before getting the base64 state from the fluid engine. 
2. VST2 Automation Handling
  - `tracksToReaperProject` should save normalized automation in `.RPP` format when included in the `track.automation` field. This will look similar (but not the same) as the `tracksToFluidMessage` code linked below. It will use `plugin.getParameterIndex` instead of `plugin.getParameterName`. 

# Notes: 

The updated `session` format allows you to write code that looks like this:

```javascript
const reverbPlugin = new fluid.DragonflyRoom();
reverbPlugin.parameters.decaySeconds = 3.0;
reverbPlugin.parameters.predelayMs = 50;
```

You can then insert `reverbPlugin` directly into a template like this
```javascript
let session = new fluid.FluidSession({
  bpm: 96,
  r: '1 + 2 + 3 + 4 + ',
  dLibrary, // default for kick and snare
  nLibrary, // default for kick and snare
}, {
  kick:  { d: '.   . mf      ', gain: -6 },
  snare: { d: 'm   f   m   f ' },
  chrd:  { nLibrary: chords.nLibrary, pan: -.75 },
  bass:  { nLibrary: { a: {type: 'midiNote', n: 36}, b: {type: 'midiNote', n: 39}, f, p } },
  tamb:  { pan: .25 },
  revb:  { plugins: [ reverbPlugin ], nLibrary: {p, q, r, s, t, u } },
});
```

This can eventually be used to build a Session object that looks like this

<img width="675" alt="Screen Shot 2020-08-25 at 1 44 16 AM" src="https://user-images.githubusercontent.com/1512520/91127567-83336600-e674-11ea-981d-caefb688cbe8.png">

(Notice that tracks are now stored in an array, and have a `.pan` and a `.gain`. )

Expanding the `plugins` array, notice the `parameters` object and the `automation` object.  These contain non-normalized values. 

<img width="676" alt="Screen Shot 2020-08-25 at 1 47 00 AM" src="https://user-images.githubusercontent.com/1512520/91127793-00f77180-e675-11ea-8603-d51b1aa92a39.png">

See how the`plugin.getParameterName(paramKey)`, and `plugin.getNormalizedValue(paramKey, explicitValue)` functions are used to get the parameter names and the normalized parameter value in the `tracksToFluidMessage` code: 

https://github.com/CharlesHolbrow/cybr/blob/ed48b02bdd22d70e7b7a1c220cf7a9a97c193320/fluid-music/src/tracksToFluidMessage.ts#L166-L195

Together, these are enough to get the VST2 base64 state AND set the automation values needed to create a reaper project.

**Comment by CharlesHolbrow** [comment]
With 653c704 TracksToReaper project should be able to access parameter indexes.

**Comment by CharlesHolbrow** [comment]
Proper VST2 state recall necessitated reverse engineering of that first base64 chunk in RPP. https://github.com/CharlesHolbrow/rppp/commits/62b2e54a54d8bb82600d6bd42d955d09f8d86e98.

**Comment by CharlesHolbrow** [comment]
`rppp v0.2.0` will support setting state chunk using the methods here:

https://github.com/CharlesHolbrow/cybr/blob/275008a174109fd9e8f0cb69d8aad4b5bb3d2ace/fluid-music/src/vst2ToReaperObject.ts#L59-L66

`initializeRouting` and `setStateChunk` are the names available in `rppp v0.2.0-1` prerelease. The API should still be considered unstable at this point. However, at this point, both parameter automation and state recall are working.
